src/components/volunteer/VolunteerFormCalendar.tsx

```diff
@@ -70,10 +73,39 @@ export const VolunteerFormCalendar = ({
 
   const isFocused = useIsFocused();
 
-  const { mutate, isLoading, isError, isSuccess, data, reset } = useMutation(calendarNew);
+  const { mutateAsync, isLoading, isError, isSuccess, data, reset } = useMutation(calendarNew);
   const [createEvent] = useMainserverMutation(CREATE_EVENT_RECORDS);
-  const onSubmit = (calendarNewData: VolunteerCalendar) => {
-    mutate(calendarNewData);
+  const onSubmit = async (calendarNewData: VolunteerCalendar) => {
+    mutateAsync(calendarNewData).then(async ({ id }) => {
+      if (id) {
+        const uris: { uri: StringConstructor; mimeType: StringConstructor }[] = [];
+        const imagesParser = JSON.parse(calendarNewData.images);
+        const documentsParser = JSON.parse(calendarNewData.documents);
+
+        if (imagesParser.length) {
+          imagesParser.forEach(({ uri = String, mimeType = String }) => {
+            uris.push({ uri, mimeType });
+          });
+        }
+
+        if (documentsParser.length) {
+          documentsParser.forEach(({ uri = String, mimeType = String }) => {
+            uris.push({ uri, mimeType });
+          });
+        }
+
+        /* foreach loop to upload each file after the event has been created */
+        uris.forEach(async ({ uri, mimeType }) => {
+          try {
+            await calendarUpload(uri, id, mimeType);
+          } catch (error) {
+            console.error(error);
+          }
+        });
```

Comment:
it ould be a bit more helpful when reading, if the order stay the same and you would initialize `uris` after `images` and `documents`. so it will be the same order from top to bottom like the for eaches.

Reply:
I think it's done now.
https://github.com/ikuseiGmbH/smart-village-app-app/pull/606/commits/9e1e9f647b49b0557f15530961d7e9028cf5c928